src/main/java/at/ac/tuwien/kr/alpha/solver/DefaultSolver.java

```diff
@@ -167,9 +174,18 @@ protected boolean tryAdvance(Consumer<? super AnswerSet> action) {
 			ConflictCause conflictCause = store.propagate();
 			didChange |= store.didPropagate();
 			LOGGER.trace("Assignment after propagation is: {}", assignment);
-			if (!disableNoGoodDeletion && conflictCause == null) {
-				// Run learned NoGood deletion strategy.
-				store.cleanupLearnedNoGoods();
+			if (conflictCause == null) {
+				if (!disableNoGoodDeletion) {
+					// Run learned NoGood deletion strategy.
+					store.cleanupLearnedNoGoods();
+				}
+				if (restartStrategy != null && restartStrategy.shouldRestart()) {
+					// Restart if necessary.
+					LOGGER.trace("Restarting search.");
+					choiceManager.backjump(0);
+					restartStrategy.onRestart();
+					continue;
+				}
 			}
 			if (conflictCause != null) {
```

Comment:
Because that `if` is followed by a chain of else-if statements., i.e., exactly only one of them is executed. The added `if (conflictCause == null)` should be executed independently of the if-else-if-... conditions.
If we were to turn that `if` in line 190 into an else-if, then the above block would prevent all other parts of the solver from being run.